src/v/cluster/controller_backend.cc

```diff
@@ -609,6 +611,8 @@ ss::future<std::error_code> controller_backend::process_partition_update(
               "creating partition {} from shard {}",
               ntp,
               previous_shard);
+            co_await raft::details::move_persistent_state(
+              requested.group, *previous_shard, ss::this_shard_id(), _storage);
```

Comment:
what failure scenarios should we be thinking about here? partition movement across cores and partition state from kvstore movement across cores doesn't happen atomically (I'm guessing) so it seems there are some crash scenarios that produce bad states?

Reply:
There are two failure scenarios that we should consider: 
1) failure before state is removed from source core
2) failure after state is removed from source core
3) failure after partial removal of state at source core

In both cases we will retry. `move_persistent_state` is idempotent and handles each piece of raft persistent state separately. It is safe to retry in all the cases, in `move_persistent_state` state is first copied to target core and only then removed. When no state is read from source core the whole operation is noop.

Reply:
how do these states interact with the process of moving the partition? for instance, if the partition is moved to the other core, but a failure happens before the state in the kvstore is moved, then what happens on restart?

Reply:
state move will be retried